src/herder/HerderImpl.cpp

```diff
@@ -450,7 +460,40 @@ TransactionQueue::AddResult
 HerderImpl::recvTransaction(TransactionFrameBasePtr tx, bool submittedFromSelf)
 {
     ZoneScoped;
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+    TransactionQueue::AddResult result;
+
+    // Allow txs of the same kind to reach the tx queue in case it can be
+    // replaced by fee
+    bool hasSoroban =
+        mSorobanTransactionQueue.sourceAccountPending(tx->getSourceID()) &&
+        !tx->isSoroban();
+    bool hasClassic =
+        mTransactionQueue.sourceAccountPending(tx->getSourceID()) &&
+        tx->isSoroban();
+    bool reject = mApp.getConfig().LIMIT_TX_QUEUE_SOURCE_ACCOUNT &&
```

Comment:
Yes, given that having two queues breaks any assumptions about source account tx chains, we should disallow core to run protocol 20 without the flag. We should enforce it prior to protocol 20 upgrade so that we don't need to drop any chains at upgrade time (we're planning to do this in the upcoming proto 19 core release). For now, to force it on futurenet, I'm going to add a check on startup behind the vNext flag.